File: agent/CarRentalPortal.Core/Entities/ShoppingCart.cs
using System.Collections.Generic;

namespace CarRentalPortal.Core.Entities
{
    public class ShoppingCart
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public int NumberOfItems { get; set; }
        public ICollection<ShoppingCartItem> Items { get; set; }
    }
}

File: agent/RentalApp/src/app/shopping-cart/shopping-cart.ts
export interface ShoppingCart {
	id?: number;
	userId: number;
	userFirstName?: string;
	numberOfItems?: number;
	items?: ShoppingCartItem[];
}

export interface ShoppingCartItem {
	id?: number;
	carId: number;
	ownerId: number;
	shoppingCartId: number;
	status?: OrderStatus;
	isBundle?: boolean;
}

export interface UpdateShoppingCart {
	id: number;
}

export enum OrderStatus {
	PENDING = 'Pending',
	ORDERED = 'Ordered'
}
